### ISSUE TITLE
Change behavior of ManagedSet.haveAtLeastOneWhere

### ISSUE THREAD

**Issue by joeconwaystk** [stablekernel/aqueduct#481]
This is getting hung by by stricter type checks, and wasn't the best solution in the first place. The preferred solution would be have variants of this expression behavior, and to also remove its signature from `ManagedSet` because it is misleading. 

Preliminary research has shown that a preferable API might be the following:

```dart
query.where((o) => o.children).have(anyOf: (child) => child.field).equalTo(...);
query.where((o) => o.children).have(noneOf: (child) => child.field).equalTo(...);
query.where((o) => o.children).have(allOf: (child) => child.field).equalTo(...);
```

Issues are that `QueryExpression<T, U>` expects to retain the type argument `U` (the object type being queried) throughout the expression chain, but this may be worth looking at if it is even viable.

**Comment by illia-romanenko** [comment]
Hi @joeconwaystk  - so as far as I understand 

> haveAtLeastOneWhere

 has been removed and there is no replacement for it at the moment? If that's so then the doc probably should be updated https://aqueduct.io/docs/db/advanced_queries/

**Comment by BlackFox1197** [comment]
Im now working with aqueduct for 2 Months, and first i really liked it, because it makes much things really easy. Of course there isn't much support for it, because it's not good known yet. But after some while, comming to some advanced queries, it really broke my neck, because most of the sql functions arent there, and some things are just impossible to do. The worst part is really, that the documentation isn't up to date at many points. Its really sad. I think its OK to not to be able on such a project the full time, but if its dead, than please mark it as such, so that people as me, don't even start working with it...

**Comment by kovac777** [comment]
@joeconwaystk Hello! I very need an example of usage without haveAtLeastOneWhere

**Comment by jayjah** [comment]
Example: 
```dart
var query = new Query<Team>(context)
  ..where((t) => t.players.haveAtLeastOneWhere.yearsPlayed).lessThanOrEqualTo(1);

var teamsWithRookies = await query.fetch();
```
- [https://aqueduct.io/docs/snippets/orm/](Aqueduct)

**Comment by kovac777** [comment]
@jayjah Yes, but haveAtLeastOneWhere was removed and now I don't see alter solutions
